File: Daily Challenge/February 2025/Minimum Operations to Exceed Threshold Value II.cpp
class Solution {
public:
  int minOperations(vector<int>& nums, int k) {
    priority_queue<long long, vector<long long>, greater<long long>> pq;

    for (int x : nums) {
      pq.push(x);
    }

    int minOperations = 0;
    while (pq.size() > 1 and pq.top() < k) {
      long long a = pq.top();
      pq.pop();
      long long b = pq.top();
      pq.pop();

      long long c = min(a, b) * 2 + max(a, b);
      pq.push(c);
      minOperations++;
    }

    return minOperations;
  }
};

File: Daily Challenge/February 2025/Minimum Operations to Exceed Threshold Value II.cs
public class Solution {
  public int MinOperations(int[] nums, int k) {
    PriorityQueue<long, long> pq = new PriorityQueue<long, long>();

    foreach (int x in nums) {
      pq.Enqueue(x, x);
    }

    int operations = 0;

    while (pq.Count > 1 && pq.Peek() < k) {
      long x = pq.Dequeue();
      long y = pq.Dequeue();
      long sum = Math.Min(x, y) * 2 + Math.Max(x, y);
      pq.Enqueue(sum, sum);
      operations++;
    }

    return operations;
  }
}